.github/actions/manage-branch/action.yml

```diff
@@ -0,0 +1,28 @@
+name: 'Manage GitHub branches'
+description: 'Manage the given GitHub branches'
+inputs:
+  branch:
+    description: 'The GitHub branch'
+    required: true
+  lock:
+    description: 'Whether to lock the branch or no'
+    required: true
+  token:
+    description: 'The GitHub access token.'
+    required: true
+  owner:
+    description: 'The GitHub owner'
+    default: ${{ github.repository }}
+  repo:
+    description: 'The GitHub repo'
+    default: ${{ github.repository }}
+  fail-on-warning:
+    description: 'Whether to fail if any warning'
+    default: false
+runs:
+  using: "composite"
+  steps:
+    - run: python ${{ github.action_path }}/lock.py ${{ inputs.owner }} ${{ inputs.repo }} ${{ inputs.branch }} ${{ inputs.lock }} ${{ inputs.fail-on-warning }}
```

Comment:
it is better to use environment variables and put them between double quotes

Reply:
yep, bear in mind, this PR is in draft and was not intended to be even in review, it was only created to illustrate that we could use this approach for freezing/unfreezing branches on demand.